include/kernels/dslash_clover_helper.cuh

```diff
@@ -42,18 +43,22 @@ namespace quda {
     real a2_minus_b2;
     QudaTwistGamma5Type twist;
 
-    CloverArg(ColorSpinorField &out, const ColorSpinorField &in, const CloverField &clover,
-	      int parity, real kappa=0.0, real mu=0.0, real epsilon = 0.0,
+    CloverArg(cvector_ref<ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in, const CloverField &clover,
+	      int parity, real kappa = 0.0, real mu = 0.0, real epsilon = 0.0,
 	      bool dagger = false, QudaTwistGamma5Type twist = QUDA_TWIST_GAMMA5_INVALID) :
-      kernel_param(dim3(in.TwistFlavor() == QUDA_TWIST_NONDEG_DOUBLET ? in.VolumeCB()/2 : in.VolumeCB(),
-                        in.TwistFlavor() == QUDA_TWIST_NONDEG_DOUBLET ? 2 : 1, in.SiteSubset())),
-      out(out), in(in),
+      kernel_param(dim3(in.TwistFlavor() == QUDA_TWIST_NONDEG_DOUBLET ? in[0].VolumeCB() / 2 : in[0].VolumeCB(),
```

Comment:
This is applicable in multiple files, but: should we add a specialized `VolumeCB()` function to `vector_ref` and get rid of explicit indexing?

Reply:
e679e8216